src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -106,24 +131,132 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
                 unsupportedBrowserDetected: () => {
                     deferred.reject(new UnsupportedBrowserError());
                 },
-                cardNonceResponseReceived: (errors, nonce) => {
-                    this._cardNonceResponseReceived(errors, nonce);
+                cardNonceResponseReceived: (errors: NonceGenerationError[] | undefined, nonce: string | undefined, cardData: CardData | undefined,
+                                            billingContact: Contact | undefined, shippingContact: Contact | undefined) => {
+                    if (errors) {
+                        this._handleNonceGenerationErrors(errors);
```

Comment:
it is a good practice to `return` here. You never know when your error handling is going to stop throwing.

Reply:
👍